### PR TITLE
Remove gtest & gmock files from installers

### PR DESCRIPTION
**Pull request by rupal-bq**
*Issue #61*

*Description of changes:*
- Added option to build with/without testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -114,18 +116,22 @@ elseif(UNIX)
 							)
 endif()
 
+if(BUILD_WITH_TESTS)
 # GTest import
```
Comment (jordanw-bq) [inline src/CMakeLists.txt:120]: add indents for `if` blocks

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -114,18 +116,22 @@ elseif(UNIX)
...
     ${PROJECT_BINARY_DIR}/googletest
     )
 enable_testing()
+endif()
 
 # Projects to build
 add_subdirectory(${ESODBC_SRC})
 add_subdirectory(${ESENLIST_SRC})
+add_subdirectory(${INSTALL_SRC})
+if(BUILD_WITH_TESTS)
```
Comment (jordanw-bq) [inline src/CMakeLists.txt:133]: indent this too

File: BUILD_INSTRUCTIONS.md [hunk 3]
```diff
@@ -38,6 +38,12 @@ Open the project's root directory in a command line interface of your choice. Ex
 
 **Note:** It is desirable to not run cmake directly in the 'src' directory, because it will generate build files inline with code.
 
+### General CMake Options
+
+**BUILD_WITH_TESTS**
+
+(Defaults to ON) If disabled, all tests will be excluded from build which will optimize the installer package size. This option can set with the command line via -D.
```
Comment (jordanw-bq) [inline BUILD_INSTRUCTIONS.md:45]: Maybe add a bit about dependencies to make it clearer
```suggestion
(Defaults to ON) If disabled, all tests and test dependencies will be excluded from build which will optimize the installer package size. This option can set with the command line (using `-D`).
```

File: src/CMakeLists.txt (hunk 1 above)
Comment (rupal-bq) [reply src/CMakeLists.txt]: added

File: src/CMakeLists.txt (hunk 2 above)
Comment (rupal-bq) [reply src/CMakeLists.txt:134]: added

File: BUILD_INSTRUCTIONS.md (hunk 3 above)
Comment (rupal-bq) [reply BUILD_INSTRUCTIONS.md]: added

**Pull request merged**